### PR TITLE
Update solidity compiler

### PR DESCRIPTION
**Pull request by dtebbs**

### CODE REVIEW

**Comment by AntoineRondelet** [comment]
@dtebbs - FYI: just updated the base branch to point to `develop` now that #349 is merged.

File: client/test_contracts/test_groth16_altbn128_mixer_base.py [hunk 1]
```diff
@@ -99,7 +99,8 @@
     zksnark_name="GROTH16",
     pairing_parameters=test_pairing.ALT_BN128_PAIRING)
 
-VK_HASH = 1
+# Arbitrary VK hash (array of 2 evm words)
+VK_HASH = [2, 1]
```
Comment (AntoineRondelet) [inline client/test_contracts/test_groth16_altbn128_mixer_base.py:103]: Wasn't that added as part of a previous PR: https://github.com/clearmatics/zeth/pull/349/files#diff-76b9f071384ce0020ab653c96c60f40beb736f4ef308d5dee9348d590fa49c77R102-R103 ?

Have you rebased on top of the latest commit on `develop` @dtebbs ?

Comment (AntoineRondelet) [reply client/test_contracts/test_groth16_altbn128_mixer_base.py:103]: (seems like a similar question holds for some of the python files below)

**Review by AntoineRondelet — COMMENTED** [review]
LGTM (besides the comments w.r.t. duplicated python changes which may be due to rebasing issues)

File: client/test_contracts/test_groth16_altbn128_mixer_base.py (hunk 1 above)
Comment (dtebbs) [reply client/test_contracts/test_groth16_altbn128_mixer_base.py]: Not sure exactly what happened, but seems like a rebase issue.  Should be fixed now.

Comment (dtebbs) [reply client/test_contracts/test_groth16_altbn128_mixer_base.py]: I hit some CI erros, possibly related to this wierd diff.  I think I must have pushed some changes to the ci branch *after* it was merged, or forgotten to push them.  This is fixed in #355.    (Not related to this PR, so should be OK to resolve this independently)

Comment (AntoineRondelet) [reply client/test_contracts/test_groth16_altbn128_mixer_base.py]: Ok I see. Weird, not sure what we did wrong here.

>  Should be fixed now.

Great, thanks

**Review by AntoineRondelet — APPROVED** [review]
LGTM, thanks for migrating to solidity 0.8 @dtebbs that's great.

**Comment by AntoineRondelet** [comment]
This PR fixes: https://github.com/clearmatics/zeth/issues/149 and https://github.com/clearmatics/zeth/issues/96

**Pull request merged**